massa-network-worker/src/peer_info_database.rs

```diff
@@ -336,6 +336,78 @@ impl PeerInfoDatabase {
         Ok(())
     }
 
+    pub async fn whitelist(&mut self, ips: Vec<IpAddr>) -> Result<(), NetworkError> {
+        for ip in ips.into_iter() {
+            let old_pt = if let Some(peer) = self.peers.get_mut(&ip) {
+                let pt = peer.peer_type;
+                if pt != PeerType::WhiteListed {
+                    peer.peer_type = PeerType::WhiteListed;
+                }
+                pt
+            } else {
+                let mut p = PeerInfo::new(ip, false);
+                p.peer_type = PeerType::WhiteListed;
+                self.peers.insert(ip, p);
+                PeerType::WhiteListed
+            };
+
+            if old_pt == PeerType::WhiteListed {
+                continue;
+            }
+            // update global connection counts by peer type
+            let peer = *self.peers.get(&ip).unwrap(); // filled just before
+            if peer.active_out_connection_attempts > 0 {
+                self.decrease_global_active_out_connection_attempt_count(old_pt, &ip)?;
+                self.increase_global_active_out_connection_attempt_count(
+                    PeerType::WhiteListed,
+                    &ip,
+                )?
+            }
+            if peer.active_out_connections > 0 {
+                self.decrease_global_active_out_connection_count(old_pt, &ip)?;
+                self.increase_global_active_out_connection_count(PeerType::WhiteListed)?
+            }
+            if peer.active_in_connections > 0 {
+                self.decrease_global_active_in_connection_count(old_pt, &ip)?;
+                self.increase_global_active_in_connection_count(PeerType::WhiteListed)?
+            }
+        }
+        self.update()
+    }
+
+    pub async fn remove_from_whitelist(&mut self, ips: Vec<IpAddr>) -> Result<(), NetworkError> {
+        for ip in ips.into_iter() {
+            let old_pt = if let Some(peer) = self.peers.get_mut(&ip) {
+                let old = peer.peer_type;
+                peer.peer_type = Default::default();
+                old
+            } else {
+                return Ok(());
+            };
+
+            if old_pt != Default::default() {
+                // update global connection counts by peer type
+                let peer = *self.peers.get(&ip).unwrap(); // filled just before
+                if peer.active_out_connection_attempts > 0 {
+                    self.decrease_global_active_out_connection_attempt_count(old_pt, &ip)?;
+                    self.increase_global_active_out_connection_attempt_count(
```

Comment:
I don't understand why you are updating the count for the default of peertype (which is standard if I remember well) if you filtered just above that type of peertype. Maybe worth some comments on why you filter to edit the counts and why you edit the default ones.

Reply:
because the peer isn't whitelist anymore, so as it became standard, we need to update the global counters

Reply:
Remove node_id from whitelist.